Registry Auth OPA Parser Can't handle Neg

**Describe the bug**

This policy will make parser throw an exception:

https://github.com/TerriaJS/magda/blob/16c2b67b7e78a45f59f3e6bf68f920a37e02543e/deploy/helm/magda/charts/opa/policies/object/registry/record/esri_groups.rego#L5

due to the `!=`

`!=` is not an operator I guess it could be converted to `Negation` in AST:
```
not input.object.registry.record["esri-access-control"].access = "private"
```

Probably can dump the actual AST to find out more.

```
[ERROR] [08/18/2020 00:52:58.735] [registry-api-akka.actor.default-dispatcher-4] [Api(akka://registry-api)] Exception encountered
java.lang.Exception: Could not understand Neq
	at au.csiro.data61.magda.registry.SqlHelper$.convertToSql(SqlHelper.scala:158)
	at au.csiro.data61.magda.registry.SqlHelper$.$anonfun$opaQueriesToWhereClauseParts$1(SqlHelper.scala:248)
	at scala.collection.immutable.List.map(List.scala:286)
	at au.csiro.data61.magda.registry.SqlHelper$.opaQueriesToWhereClauseParts(SqlHelper.scala:194)
	at au.csiro.data61.magda.registry.SqlHelper$.$anonfun$getOpaConditions$2(SqlHelper.scala:60)
	at scala.collection.immutable.List.map(List.scala:286)
	at au.csiro.data61.magda.registry.SqlHelper$.$anonfun$getOpaConditions$1(SqlHelper.scala:57)
	at scala.collection.immutable.List.flatMap(List.scala:338)
	at au.csiro.data61.magda.registry.SqlHelper$.getOpaConditions(SqlHelper.scala:50)
	at au.csiro.data61.magda.registry.DefaultRecordPersistence.aspectIdsToSelectClauses(RecordPersistence.scala:1804)
	at au.csiro.data61.magda.registry.DefaultRecordPersistence.getRecords(RecordPersistence.scala:1501)
	at au.csiro.data61.magda.registry.DefaultRecordPersistence.getByIdWithAspects(RecordPersistence.scala:410)
	at au.csiro.data61.magda.registry.RecordsServiceRO.$anonfun$getById$7(RecordsServiceRO.scala:823)
	at scalikejdbc.DBConnection.readOnly(DBConnection.scala:202)
	at scalikejdbc.DBConnection.readOnly$(DBConnection.scala:200)
	at scalikejdbc.DB.readOnly(DB.scala:60)
	at scalikejdbc.DB$.$anonfun$readOnly$1(DB.scala:175)
	at scalikejdbc.LoanPattern.using(LoanPattern.scala:18)
	at scalikejdbc.LoanPattern.using$(LoanPattern.scala:16)
	at scalikejdbc.DB$.using(DB.scala:140)
	at scalikejdbc.DB$.readOnly(DB.scala:174)
	at au.csiro.data61.magda.registry.RecordsServiceRO.$anonfun$getById$6(RecordsServiceRO.scala:822)
	at akka.http.scaladsl.server.util.ApplyConverterInstances$$anon$1.$anonfun$apply$1(ApplyConverterInstances.scala:14)
	at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$tprovide$1(BasicDirectives.scala:143)
	at akka.http.scaladsl.server.Directive$$anon$1.tapply(Directive.scala:125)
	at akka.http.scaladsl.server.Directive.$anonfun$tflatMap$2(Directive.scala:69)
	at akka.http.scaladsl.server.directives.OnSuccessMagnet$$anon$1.$anonfun$directive$3(FutureDirectives.scala:97)
	at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41)
	at akka.http.scaladsl.util.FastFuture$.$anonfun$transformWith$3(FastFuture.scala:51)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:92)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:92)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:49)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -123,6 +123,7 @@ Registry:
 
 -   Made events record the correct user id
 -   Fixed Registry History API Performance Issue when limit=1 & Updated Registry History API Document
+-   Fixed not support `Neg` operator in OPA policy
 
 Search:
 

--- a/magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/SqlHelper.scala
+++ b/magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/SqlHelper.scala
@@ -154,6 +154,7 @@ object SqlHelper {
     else if (operation == Lt) SQLSyntax.createUnsafely("<")
     else if (operation == Gte) SQLSyntax.createUnsafely(">=")
     else if (operation == Lte) SQLSyntax.createUnsafely("<=")
+    else if (operation == Neq) SQLSyntax.createUnsafely("!=")
     else
       throw new Exception("Could not understand " + operation)
   }

